pkg/deviceplugin/server.go

```diff
@@ -141,6 +142,12 @@ func (srv *server) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequest)
 			for i := range dev.mounts {
 				cresp.Mounts = append(cresp.Mounts, &dev.mounts[i])
 			}
+			if cresp.Annotations == nil {
+				cresp.Annotations = make(map[string]string)
+			}
+			cresp.Annotations["INTEL_PLUGIN_DEVICES"] = cresp.Annotations["INTEL_PLUGIN_DEVICES"] + sep + id
+			sep = ","
```

Comment:
The reason for this kludge was to avoid changes in the PostAllocate API and related changes in every plugin which uses it. I didn't really want to use the annotations, but that was a way to avoid API changes. Now if the API change is ok, I would opt for adding the entire AllocateRequest there, like: err := srv.postAllocate(response, rqt)
What do you think?

Reply:
Good point. Now I see the current API doesn't satisfy this use case. The initial assumption was that devnodes and mounts are immutable for a given device ID.

How about introducing a new optional interface for `Allocate()` replacing the server's `Allocate()` completely in case the plugin implements it.

Reply:
"Allocate" sounds good to me

Reply:
The plugin's Allocate can return something like UseDefaultAllocateError in case of disabled resourceManager.

Reply:
Here my take is that the cleanest solution is to modify PostAllocate and change all the plugins that use it. Second cleanest would be to add PostAllocateV2. Now I added "Allocate" which is basically PostAllocateV2 and I don't like it. But having a full replacement of the Allocate functionality, is just too messy in the plugin side of things. And unnecessary.